core/auth.py

```diff
@@ -480,17 +489,26 @@ def api_validateUser():
                     data = json.loads(jimi.api.request.data)
                     #check if OTP has been passed
                     #if invalid user or user requires OTP, return 200 but request OTP
+                    userSession = None
+                    #Check if there's only one login type available
+                    loginTypes = jimi.settings.getSettingValue(None,jimi.api.g.sessionData,"auth","types")
+                    if len(loginTypes) == 1:
+                        data["type"] = loginTypes[0]
                     if "otp" not in data:
                         user = _user().getAsClass(query={ "username" : data["username"] })
                         if len(user) == 1:
                             user = user[0]
                             if user.totpSecret != "":
                                 return {}, 403
                             else:
-                                userSession = validateUser(data["username"],data["password"])
+                                if data["type"] == "local":
+                                    userSession = validateUser(data["username"],data["password"])
+                                else:
+                                    userSession = validateExternalUser(data["username"],data["password"],data["type"],application="jimi",userData=user)
```

Comment:
Does this mean that the login type for a given user is defined by the login form? Do we want to change this so that user type is loaded based on username provided? I assume that authorisation is still done by jimi?

Reply:
I've updated jimi to use the user's "loginType" value instead, which makes more sense for now. We can probably look in the future about moving away from 1to1 mappings.